plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -650,6 +655,43 @@ public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHan
         }
     }
 
+    @Override
+    public JdbcProcedureHandle getProcedureHandle(ConnectorSession session, ProcedureInformation procedureInformation)
+
+    {
+        if (procedureInformation.inputArguments().stream().anyMatch(argument -> argument.contains(";"))) {
```

Comment:
If we can process arguments via `preparedCall.setParameter` we wouldn't need this.
if we cannot process  arguments via `preparedCall.setParameter` we shouldn't pretend we can. 
If the user provides parameter values as string literals to be concatenated with the procedure call, why can't they provide the whole procedure call (including the parameters)? Just like we do with SQL pushdown?

Reply:
> If we can process arguments via preparedCall.setParameter we wouldn't need this.

It could be possible - like if we could get the procedure column information and map them accordingly. I would like implement them in an incremental way as executing stored procedure is kind of an experimental feature. WDYT ?

> why can't they provide the whole procedure call (including the parameters)? Just like we do with SQL pushdown?

The initial approach we have is like SQL Pushdown - where we don't do any processing but consume a `String` as an input. For some datasources like `SQLServer` it would be helpful if we could get the schema and procedure name - so we could apply some checks to ensure the SP doesn't any DDL or DML operations. So it more of an extended version of SQLPushdown - where we take some of them in parts and concat all the pieces.